### PR TITLE
Abdm facility fetch changes

### PR DESCRIPTION
**Pull request by helenKaryamsetty**
## 📋 Description

JIRA ID: AMM-871

Changes to fetch mapped ABDM Facility Id from job location.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [x] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

- **New Features**
	- Introduced a REST controller for managing ABDM facility requests, enabling retrieval of facility details based on work location.
	- Added new fields to enhance the data model for storing ABDM facility identifiers.
	- Implemented a repository interface for custom queries related to ABDM facilities.
	- Created a service interface and its implementation for accessing mapped ABDM facilities.

- **Bug Fixes**
	- Improved error handling in the facility retrieval process.

### CODE REVIEW

**Comment by drtechie** [comment]
@helenKaryamsetty several comments from coderabbit including typos and function names are not addressed. Have you checked these comments?

File: src/main/java/com/iemr/common/repository/abdmfacility/AbdmFacilityRepository.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import com.iemr.common.data.users.ProviderServiceAddressMapping;
+
+@Repository
+public interface AbdmFacilityRepository extends CrudRepository<ProviderServiceAddressMapping, Integer> {
+	
+	@Query("SELECT v from ProviderServiceAddressMapping v where v.pSAddMapID=:pssmID order by 1 desc")
+	ProviderServiceAddressMapping getAbdamFacility(@Param("pssmID") int pssmID);
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/repository/abdmfacility/AbdmFacilityRepository.java:14]: i want to keep the ordering it is required

File: src/main/java/com/iemr/common/controller/abdmfacility/AbdmFacilityController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+			String resp = abdmFacilityService.getMappedAbdmFacility(workLocationId);
+
+				response.setResponse(resp);
+
+		} catch (Exception e) {
+
+			response.setError(5000, e.getMessage());
+			logger.error(e.getMessage());
+		}
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/controller/abdmfacility/AbdmFacilityController.java:45]: We are handling it in otput response, not required to give separate exception for it.

File: src/main/java/com/iemr/common/controller/abdmfacility/AbdmFacilityController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	
+	private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+	
+	@Autowired
+	private AbdmFacilityService abdmFacilityService;
+	
+	@CrossOrigin
+	@Operation(summary = "Get Abdm facility mapped to worklocation")
+	@GetMapping(value = { "/getWorklocationMappedAbdmFacility/{workLocationId}" })
+	public String getAbdmFacilityDetails(@PathVariable int workLocationId, @RequestHeader(value = "Authorization") String Authorization) {
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/controller/abdmfacility/AbdmFacilityController.java:31]: we follow this pattern for all APIs. Hence, maintaining the commanility

**Comment by helenKaryamsetty** [comment]
> @helenKaryamsetty several comments from coderabbit including typos and function names are not addressed. Have you checked these comments?

yes committed those changes

**Review by ravishanigarapu — APPROVED** [review]
looks  fine

**Pull request merged**